### PR TITLE
Release notes and highlights for 2.15

### PR DESCRIPTION
**Pull request by rhr323**
Adding Release notes and Release highlights for the upcoming 2.15.0 release.

### CODE REVIEW

File: docs/release-notes/2.15.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,63 @@
+:issue: https://github.com/elastic/cloud-on-k8s/issues/
+:pull: https://github.com/elastic/cloud-on-k8s/pull/
+
+[[release-notes-2.15.0]]
+== {n} version 2.15.0
+
```
Comment (rhr323) [inline docs/release-notes/2.15.0.asciidoc:6]: Should we add information about the role mapping fix in 8.16, or is that not strictly related to this ECK release?

File: docs/release-notes/highlights-2.15.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[[release-highlights-2.15.0]]
+== 2.15.0 release highlights
+
```
Comment (rhr323) [inline docs/release-notes/highlights-2.15.0.asciidoc:3]: Should we add information about the role mapping fix in 8.16 (as a known issue maybe), or is that not strictly related to this ECK release?

File: docs/release-notes/highlights-2.15.0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+[float]
+[id="{p}-2150-wolfi-images"]
+=== Wolfi images
+
+ECK has transitioned to using https://github.com/wolfi-dev[Wolfi images], an open-source project aimed at delivering a secure, minimal base image for containerized applications. By adopting Wolfi, ECK strengthens its security posture, significantly lowering the potential for CVEs (Common Vulnerabilities and Exposures) and reducing the overall attack surface. This shift aligns with best practices for container security, providing a more robust foundation for running workloads reliably and securely.
+
+[float]
+[id="{p}-2150-latest-k8s-support"]
+=== Support for Kubernetes 1.31
```
Comment (rhr323) [inline docs/release-notes/highlights-2.15.0.asciidoc:18]: I’m not sure if it’s worth mentioning here, since we didn’t make any changes specific to the latest K8s release.

File: docs/release-notes/2.15.0.asciidoc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+[[enhancement-2.15.0]]
+[float]
+=== Enhancements
+
+* Clarify high availability recommendations in Elasticsearch orchestration docs {pull}8151[#8151]
```
Comment (barkbay) [inline docs/release-notes/2.15.0.asciidoc:14]: I think this one should be in the `Documentation improvements` section?

File: docs/release-notes/2.15.0.asciidoc [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+* Log when k8s resources are created/updated/deleted successfully {pull}8094[#8094]
+* More meaningful error in readiness script {pull}8091[#8091] (issue: {issue}8088[#8088])
+* Update Maps Images to multi-arch {pull}8085[#8085] (issue: {issue}8034[#8034])
+* Add missing metricsets {pull}8048[#8048] (issue: {issue}7277[#7277])
```
Comment (barkbay) [inline docs/release-notes/2.15.0.asciidoc:18]: Let's maybe clarify what metricset, and that this is in the context of stack monitoring?

File: docs/release-notes/2.15.0.asciidoc [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+* Do not parse the version twice {pull}8005[#8005]
```
Comment (barkbay) [inline docs/release-notes/2.15.0.asciidoc:19]: I think this one is a refactoring, not really a feature enhancement?

File: docs/release-notes/highlights-2.15.0.asciidoc (hunk 3 above)
Comment (barkbay) [reply docs/release-notes/highlights-2.15.0.asciidoc:18]: Yes, we didn't usually mention this.

Comment (pebrc) [reply docs/release-notes/highlights-2.15.0.asciidoc:18]: +1 for leaving this out to stay consistent with previous messaging.

Comment (rhr323) [reply docs/release-notes/highlights-2.15.0.asciidoc:18]: Thanks for the feedback. I'll remove this part from the highlights.

If we simplify the Wolfi adoption section, maybe we could position this as a ‘maintenance’ release focused on enhancements, documentation improvements, and dependency updates. This would make the release highlights shorter than usual, without calling out specific changes. What do you think, @pebrc, @barkbay?

**Comment by rhr323** [comment]
Hey @pebrc , @barkbay ,

I have simplified the release highlights to a single sentence. Let me know what you think. Also updated the release notes according to the feedback received. Thanks!

File: docs/release-notes/2.15.0.asciidoc [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+* Move to Wolfi-based images {pull}7977[#7977]
+
+[[bug-2.15.0]]
+[float]
+=== Bug fixes
+
+* Ensure esclient is closed after each reconcile. {pull}8175[#8175] (issue: {issue}8174[#8174])
+* Fix resetting service type to default when not specified {pull}8165[#8165] (issue: {issue}8161[#8161])
+* Fix logstash templating issue in Helm chart {pull}8087[#8087] (issue: {issue}8000[#8000])
+* Move '$leading_path' variable definition in eck-operator {pull}8075[#8075]
```
Comment (pebrc) [inline docs/release-notes/2.15.0.asciidoc:27]:
```suggestion
* Move '$leading_path' variable definition in eck-operator Helm chart {pull}8075[#8075]
```

File: docs/release-notes/2.15.0.asciidoc [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+* Handle Kibana basepath updates for fleet {pull}8053[#8053] (issue: {issue}7909[#7909])
```
Comment (pebrc) [inline docs/release-notes/2.15.0.asciidoc:28]:
```suggestion
* Support Kibana basepath in associations {pull}8053[#8053] (issue: {issue}7909[#7909])
```

File: docs/release-notes/2.15.0.asciidoc [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+* Fix eck-stack kibana examples in Helm chart {pull}8041[#8041]
```
Comment (pebrc) [inline docs/release-notes/2.15.0.asciidoc:29]:
```suggestion
* Fix eck-stack Kibana examples in Helm chart {pull}8041[#8041]
```

File: docs/release-notes/2.15.0.asciidoc [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+* Add watcher for statefulsets in agent controller {pull}8011[#8011]
```
Comment (pebrc) [inline docs/release-notes/2.15.0.asciidoc:30]:
```suggestion
* Add watcher for StatefulSets in Elastic Agent controller {pull}8011[#8011]
```

File: docs/release-notes/highlights-2.15.0.asciidoc (hunk 2 above)
Comment (pebrc) [reply docs/release-notes/highlights-2.15.0.asciidoc:3]: I would probably still include the existing known issue. And we need to amend it to mention that the issue is resolved in 8.16. We can do that in a separate PR though.

**Review by pebrc — APPROVED** [review]

**Comment by rhr323** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.15|[](https://github.com/elastic/cloud-on-k8s/pull/8191)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)
